components/clp-package-utils/clp_package_utils/scripts/archive_manager.py

```diff
@@ -0,0 +1,227 @@
+        return False
+    if end_ts is not None and end_ts < 0:
+        logger.error("end-ts must be non-negative.")
+        return False
+    if begin_ts is not None and end_ts is not None and begin_ts > end_ts:
+        logger.error("begin-ts must be <= end-ts")
+        return False
+    return True
+
+
+def main(argv):
+    clp_home = get_clp_home()
+    default_config_file_path = clp_home / CLP_DEFAULT_CONFIG_FILE_RELATIVE_PATH
+
+    # Top-level parser and options
+    args_parser = argparse.ArgumentParser(description="View or delete archives.")
+    args_parser.add_argument(
+        "--config",
+        "-c",
+        default=str(default_config_file_path),
+        help="CLP package configuration file.",
+    )
+
+    # Top-level commands
+    subparsers = args_parser.add_subparsers(
+        dest="subcommand",
+        required=True,
+    )
+    find_parser = subparsers.add_parser(
+        FIND_COMMAND,
+        help="List IDs of archives.",
+    )
+    del_parser = subparsers.add_parser(
+        DEL_COMMAND,
+        help="Delete archives.",
+    )
+
+    # Options for find subcommand
+    find_parser.add_argument(
+        BEGIN_TS_ARG,
+        dest="begin_ts",
+        type=int,
+        default=0,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    find_parser.add_argument(
+        END_TS_ARG,
+        dest="end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    # Options for delete subcommand
+    del_parser.add_argument(
+        DRY_RUN_ARG,
+        dest="dry_run",
+        action="store_true",
+        help="Only prints the archives to be deleted, without actually deleting them.",
+    )
+
+    # Subcommands for delete subcommand
+    del_subparsers = del_parser.add_subparsers(
+        dest="del_subcommand",
+        required=True,
+    )
+
+    # Delete by ID subcommand
+    del_id_parser = del_subparsers.add_parser(
+        BY_IDS_COMMAND,
+        help="Delete archives by ID.",
+    )
+
+    # Delete by ID arguments
+    del_id_parser.add_argument(
+        "ids",
+        nargs="+",
+        help="List of archive IDs to delete",
+    )
+
+    # Delete by filter subcommand
+    del_filter_parser = del_subparsers.add_parser(
+        BY_FILTER_COMMAND,
+        help="Delete archives that fall within the specified time range.",
+    )
+
+    # Delete by filter arguments
+    del_filter_parser.add_argument(
+        BEGIN_TS_ARG,
+        type=int,
+        default=0,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    del_filter_parser.add_argument(
+        END_TS_ARG,
+        type=int,
+        required=True,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    begin_timestamp = None
+    end_timestamp = None
```

Comment:
Nit:
```suggestion
    begin_timestamp: int
    end_timestamp: int
```

They should get properly None initialized.